packages/agw-client/src/actions/linkToAgw.ts

```diff
@@ -0,0 +1,183 @@
+import { BaseError } from 'abitype';
+import {
+  type Account,
+  type Address,
+  type Chain,
+  createPublicClient,
+  decodeEventLog,
+  encodeFunctionData,
+  type Hash,
+  http,
+  type PublicClient,
+  type Transport,
+  type WalletClient,
+} from 'viem';
+import { writeContract } from 'viem/actions';
+import { getAction, parseAccount } from 'viem/utils';
+import { publicActionsL2 } from 'viem/zksync';
+
+import { BridgeHubAbi } from '../abis/BridgeHubAbi.js';
+import { DelegateRegistryAbi } from '../abis/DelegateRegistry.js';
+import { ZkSyncAbi } from '../abis/ZkSyncAbi.js';
+import {
+  BASE_GAS_PER_PUBDATA_BYTE,
+  BRIDGEHUB_ADDRESS,
+  CANONICAL_DELEGATE_REGISTRY_ADDRESS,
+  NON_EXPIRING_DELEGATION_RIGHTS,
+} from '../constants.js';
+import { AccountNotFoundError } from '../errors/account.js';
+import { VALID_CHAINS } from '../utils.js';
+
+export interface LinkToAgwParameters {
+  agwAddress: Address;
+  enabled: boolean;
+  l2Chain: Chain;
+  account?: Account;
+}
+
+export interface LinkToAgwReturnType {
+  l1TransactionHash: Hash;
+  getL2TransactionHash: () => Promise<Hash>;
+}
+
+export async function linkToAgw(
+  client: WalletClient<Transport, Chain, Account>,
+  parameters: LinkToAgwParameters,
+): Promise<LinkToAgwReturnType> {
+  const {
+    account: account_ = client.account,
+    agwAddress,
+    enabled,
+    l2Chain,
+  } = parameters;
+
+  if (!account_)
+    throw new AccountNotFoundError({
+      docsPath: '/docs/contract/writeContract',
```

Comment:
We probably don't want to throw this error pointing to writeContract. Not sure if there's an option without a path, but we should do that